slideshow/src/main/java/nl/ou/domain/ContentVisitor.java

```diff
@@ -0,0 +1,40 @@
+package nl.ou.domain;
+
+public interface ContentVisitor {
```

Comment:
Dit is totaal geen standaard Visitor pattern, maar op zich zie ik er geen probleem mee. Het is heel realistisch om te verwachten dat een Visitor een actie moet uitvoeren voor _en_ na de acties op de child components. Dat moeten we alleen wel aangeven in ons verslag, waarom we dat doen.
Ik zou dan de `end` methode wel opsplitsen in twee en de naamgeving wat meer consistent maken: `doForListContentStart`, `doForListContentEnd`, `doForTableContentStart` en `doForTableContentEnd`.

Reply:
Bijvoorbeeld https://docs.oracle.com/javase/8/docs/api/java/nio/file/FileVisitor.html heeft ook pre&post visits voor directories, dat lijkt me iets heel normaals om in een visitor te stoppen.

Reply:
Ah, leuk. Er is zelfs een concreet voorbeeld te vinden. Hier was ik niet mee bekend. Helemaal mee eens ook, dat het past in een visitor. Is alleen niet de standaard visitor, dus we moeten wel aangeven _dat_ we afwijken en waarom.